test/System/String.test.cpp

```diff
@@ -58,6 +58,134 @@ struct StringMaker<std::basic_string<std::uint8_t>>
+        const sf::U8StringCharTraits::char_type s2[] = {'a', 'b', 'c', '\0'};
+        const sf::U8StringCharTraits::char_type s3[] = {'d', 'e', 'f', '\0'};
+        CHECK(sf::U8StringCharTraits::compare(s1, s2, std::size(s1)) == 0);
+        CHECK(sf::U8StringCharTraits::compare(s1, s3, std::size(s1)) < 0);
+        CHECK(sf::U8StringCharTraits::compare(s3, s1, std::size(s3)) > 0);
+    }
+
+    SECTION("length()")
+    {
+        const sf::U8StringCharTraits::char_type s1[] = {'a', '\0'};
+        const sf::U8StringCharTraits::char_type s2[] = {'a', 'b', 'c', 'd', 'e', '\0'};
+        CHECK(sf::U8StringCharTraits::length(s1) == 1);
+        CHECK(sf::U8StringCharTraits::length(s2) == 5);
+    }
+
+    SECTION("find()")
+    {
+        const sf::U8StringCharTraits::char_type s[] = {'a', 'b', 'c', 'd', 'e'};
+        CHECK(*sf::U8StringCharTraits::find(s, std::size(s), 'a') == 'a');
+        CHECK(sf::U8StringCharTraits::find(s, std::size(s), 'f') == nullptr);
+    }
+
+    SECTION("to_char_type()")
+    {
+        CHECK(sf::U8StringCharTraits::to_char_type(sf::U8StringCharTraits::int_type{0}) ==
+              sf::U8StringCharTraits::char_type{0});
+        CHECK(sf::U8StringCharTraits::to_char_type(sf::U8StringCharTraits::int_type{1}) ==
+              sf::U8StringCharTraits::char_type{1});
+        CHECK(sf::U8StringCharTraits::to_char_type(sf::U8StringCharTraits::int_type{10}) ==
+              sf::U8StringCharTraits::char_type{10});
+    }
+
+    SECTION("to_int_type()")
+    {
+        CHECK(sf::U8StringCharTraits::to_int_type(sf::U8StringCharTraits::char_type{0}) ==
+              sf::U8StringCharTraits::int_type{0});
+        CHECK(sf::U8StringCharTraits::to_int_type(sf::U8StringCharTraits::char_type{1}) ==
+              sf::U8StringCharTraits::int_type{1});
+        CHECK(sf::U8StringCharTraits::to_int_type(sf::U8StringCharTraits::char_type{10}) ==
+              sf::U8StringCharTraits::int_type{10});
+    }
+
+    SECTION("eq_int_type()")
+    {
+        CHECK(sf::U8StringCharTraits::eq_int_type(sf::U8StringCharTraits::int_type{0}, sf::U8StringCharTraits::int_type{0}));
+        CHECK(sf::U8StringCharTraits::eq_int_type(sf::U8StringCharTraits::int_type{1}, sf::U8StringCharTraits::int_type{1}));
+        CHECK(sf::U8StringCharTraits::eq_int_type(sf::U8StringCharTraits::int_type{10},
+                                                  sf::U8StringCharTraits::int_type{10}));
+    }
+
+    SECTION("eof()")
+    {
+        CHECK(sf::U8StringCharTraits::eof() == ~sf::U8StringCharTraits::int_type{0});
+    }
+
+    SECTION("not_eof()")
+    {
+        CHECK(!sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::int_type{0}));
+        CHECK(sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::int_type{1}));
+        CHECK(sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::int_type{10}));
+        CHECK(!sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::eof()));
```

Comment:
This function is supposed to return the argument when it is not eof, let's test that:
```suggestion
        CHECK(sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::int_type{0}) == sf::U8StringCharTraits::int_type{0});
        CHECK(sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::int_type{1}) == sf::U8StringCharTraits::int_type{1});
        CHECK(sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::int_type{10}) == sf::U8StringCharTraits::int_type{10});
        CHECK(sf::U8StringCharTraits::not_eof(sf::U8StringCharTraits::eof()) != sf::U8StringCharTraits::eof());
```

Besides, not sure if being explicit about `sf::U8StringCharTraits::int_type` is useful and it is quite verbose.